cibyl/sources/jenkins.py

```diff
@@ -307,6 +311,76 @@ def get_last_build(self, **kwargs):
 
         return AttributeDictValue("jobs", attr_type=Job, value=job_objects)
 
+    @speed_index({'base': 2})
+    def get_tests(self, **kwargs):
+        """
+            Get tests for a Jenkins job.
+
+            :returns: container of jobs with the last completed build
+            (if any) and the tests
+            :rtype: :class:`AttributeDictValue`
+        """
+
+        # Get the jobs with the last completed build (default behavior)
+        jobs_found = self.send_request(
+            self.jobs_last_completed_build_query)["jobs"]
+        jobs_filtered = filter_jobs(jobs_found, **kwargs)
+
+        job_objects = {}
+
+        for job in jobs_filtered:
+            job_name = job.get('name')
+            if job["lastCompletedBuild"] is None:
+                LOG.warning("No completed builds found for job %s", job_name)
+                continue
+
+            job_object = Job(name=job_name, url=job.get('url'))
+            job_objects[job_name] = job_object
+
+            if kwargs.get('build_id'):
+                # For specific build ids we have to fetch them
+                builds = self.send_request(item=f"job/{job_name}",
+                                           query=self.jobs_builds_query.get(
+                                               kwargs.get('verbosity'), 0))
+                builds_to_add = filter_builds(builds["allBuilds"], **kwargs)
+            else:
+                builds_to_add = filter_builds([job["lastCompletedBuild"]],
+                                              **kwargs)
+
+            if not builds_to_add:
+                LOG.warning("No builds found for job %s", job_name)
+                continue
+
+            for build in builds_to_add:
+                build_object = Build(
+                    build_id=build['number'],
+                    status=build['result'])
+
+                if build['result'] == 'FAILURE':
+                    LOG.warning("Build %s for job %s failed. No tests to "
+                                "fetch", build['number'], job_name)
+                    continue
+
+                # Get the tests for this build
+                tests_found = self.send_request(
```

Comment:
This request can return 404 if a job is succesfull and contains no tests. To reproduce run `cibyl --tests --jobs designate`. The simplest solution would be to catch the JenkinsError exception, check if it contains 404 in the error message and warn. If it's a different exception, then raise.

Reply:
Done. Thanks for the tip